### PR TITLE
Milo: Automated Breadcrumbs (attempt 2)

### PR DESCRIPTION
**Pull request by jckautzmann**
- adds support for automated breadcrumbs
- automated breadcrumbs are disabled by default

Note: previous PR (https://github.com/adobecom/milo/pull/490) had been reverted as breadcrumbs were automatically generated by default: this has been fixed with https://github.com/adobecom/milo/pull/586: by default automatically generated breadcrumbs is disabled.

Pasting here all the supported features and test URLs from the original PR.

Supported ways to generate a breadcumbs in the page: 
1) breadcumbs block on a page
2) breadcumbs block in a `breadcumbs.docx` file in the same folder as the page or in an ancestor folder: the first file in the ancestor chain is taken into account. The current page title is appended to the breadcrumbs. 
3) auto-generated breadcumbs from the URL (fallback behaviour)

The first condition that applies to the page (according to order above) wins and the page renders the specified breadcrumbs.

Following page metadata properties are supported:
- `breadcrumbs-title`: set a specific page title in the breadcrumbs
- `breadcrumbs-hide-page`: set to `on` to hide the page item in the breadcrumbs
- `breadcrumbs-hide-last`: set to `on` to hide the last page item in the breadcrumbs
- `breadcrumbs`: set to `off` to not generate the breadcrumbs on a page

Resolves: [MWPW-117620](https://jira.corp.adobe.com/browse/MWPW-117620)

**Test URLs: the breadcrumbs block is defined in the current page:**
- Before: https://main--milo--adobecom.hlx.page/drafts/jck/breadcrumbs/1-from-page/test?martech=off
- After: https://breadcrumbs-auto--milo--adobecom.hlx.page/drafts/jck/breadcrumbs/1-from-page/test?martech=off

**Test URLs: the breadcrumbs block is defined in a breadcrumbs.docx file in the same folder:**
- Before: https://main--milo--adobecom.hlx.page/drafts/jck/breadcrumbs/2-from-file/test?martech=off
- After: https://breadcrumbs-auto--milo--adobecom.hlx.page/drafts/jck/breadcrumbs/2-from-file/test?martech=off

**Test URLs: the breadcrumbs block is defined in a breadcrumbs.docx file in an ancestor folder:**
- Before: https://main--milo--adobecom.hlx.page/drafts/jck/breadcrumbs/3-from-ancestor-file/level1/level2/test?martech=off
- After: https://breadcrumbs-auto--milo--adobecom.hlx.page/drafts/jck/breadcrumbs/3-from-ancestor-file/level1/level2/test?martech=off

**Test URLs: the breadcrumbs is automatically generated from the URL:**
- Before: https://main--milo--adobecom.hlx.page/drafts/jck/breadcrumbs/4-from-url/test?martech=off
- After: https://breadcrumbs-auto--milo--adobecom.hlx.page/drafts/jck/breadcrumbs/4-from-url/test?martech=off

**Test URLs: the breadcrumbs is disabled:**
* Before: https://main--milo--adobecom.hlx.page/drafts/jck/breadcrumbs/5-no-breadcrumbs/test?martech=off
* After: https://breadcrumbs-auto--milo--adobecom.hlx.page/drafts/jck/breadcrumbs/5-no-breadcrumbs/test?martech=off

**Test URLs: one breadcrumbs item has a specific breadcrumbs title:**
* Before: https://main--milo--adobecom.hlx.page/drafts/jck/breadcrumbs/6-specific-title/level1/level2/test?martech=off
* After: https://breadcrumbs-auto--milo--adobecom.hlx.page/drafts/jck/breadcrumbs/6-specific-title/level1/level2/test?martech=off

**Test URLs: one breadcrumbs item is hidden:**
* Before: https://main--milo--adobecom.hlx.page/drafts/jck/breadcrumbs/7-hide-page/level1/level2/test?martech=off
* After: https://breadcrumbs-auto--milo--adobecom.hlx.page/drafts/jck/breadcrumbs/7-hide-page/level1/level2/test?martech=off

**Test URLs: the last breadcrumbs item is hidden:**
* Before: https://main--milo--adobecom.hlx.page/drafts/jck/breadcrumbs/8-hide-last-page/test?martech=off
* After: https://breadcrumbs-auto--milo--adobecom.hlx.page/drafts/jck/breadcrumbs/8-hide-last-page/test?martech=off

### CODE REVIEW

**Review by chrischrischris — APPROVED** [review]

**Review by mokimo — CHANGES_REQUESTED** [review]
I'll block this PR requesting changes for now. This landed in the Eurovision's backlog, and we will be addressing this PR post MVP of the new navigation. Until then, there is no sense in merging this into the old navigation - it would just lead to confusion why it is available in the old navigation and not the new one.

**Comment by honstar** [comment]
Given @mokimo's reasoning that automated breadcrumbs will be taken on by the eurovision team as part of their backlog, shouldn't we rather close this PR instead of leaving it open for an unknown period of time? @jckautzmann , would that be ok?

**Pull request closed without merging**
